### ISSUE TITLE
register keyword does not work as expected with PCE target

### ISSUE THREAD

**Issue by mrdudz** [cc65/cc65#1090]
Since the default config for PCE is to map Zeropage not at $0000 (not sure why) using the register keyboard in C code will almost certainly produce code that does not work as expected.

I would suggest to either
- disable the register Keyword for this target, or
- not map the Zeropage somewhere else than $0000

(obviously fixing the compiler to produce code that works as expected is also an option, but i dont see that happening anytime soon)

**Comment by oliverschmidt** [comment]
You sort of lost me. So far I thought one would need use `-Or` to have the register keyword do anything in the first place. From that perspective users just shouldn't use `-Or` with PCE.

**Comment by mrdudz** [comment]
Using the same logic you can argue "do not use -O, the optimizer might produce broken code". No, -Or should never do something that results in broken code, IMHO.

That said, it worked in my original implementation - i am not sure why the zeropage must be mapped so it doesnt work anymore.

**Comment by polluks** [comment]
@mrdudz "The zero page/stack on the Huc6280 is located at absolute address 0x2000 instead of 0x0000 like on 6502, because they use MPR1 rather than MPR0 for their page address."
https://twitter.com/eggboycolor/status/1090115311959986177

BTW 'This has some annoying side effects in ca65.
The VDC registers were "optimized" to ZP.' :-o

**Comment by mrdudz** [comment]
Yes, but you can still map the RAM to $0000 as well, which i believe i did back then - and then everything works as expected.

**Comment by polluks** [comment]
Hmm, what about the BIOS of the CD subsystem?

**Comment by mrdudz** [comment]
what about it? what support do the libs have for this?

**Comment by polluks** [comment]
Indeed, right now nothing but what about the future?

**Comment by mrdudz** [comment]
I am not sure what we are discussing here. I am not questioning there are good reasons for doing it like it was done. 
However, there is no good reason to generate broken code when -Or is used and not telling the user why that is the case. https://cc65.github.io/doc/pce.html is only listing the memory layout, but fails to explain why this specific layout was choosen, and implications that has. If you dont already know the target very well, then it will be very surprising behaviour that enabling -Or results in random I/O writes.

**Comment by polluks** [comment]
Well, I was not talking about the register issue (full ack), but "you can still map the RAM to $0000". I suppose we would lose BIOS compatibility (just guessing).

**Comment by mrdudz** [comment]
Maybe, mabe not. You can change the banking at any time (and back).
